### PR TITLE
Add documentation summary for Iri's

### PR DESCRIPTION
**Pull request by runewake2**
This is a tiny doc addition and may not be complete. Educating myself about Iri's and adding resources and notes here to help others. Feel free to close if taking docs in another direction seems like a better option.

### CODE REVIEW

File: Letterbook.ActivityPub/CompactIri.cs
Comment (jenniferplusplus) [inline Letterbook.ActivityPub/CompactIri.cs:1]: I'm happy to have inline docs. Just a couple of things to add.
1. I'm not entirely sure my implementation supports IRIs beyond the ascii set. I know there are parsing options for that, but if they're not used by default then we'll need to update to use them at some point.
2. JSON-LD lets you define named arbitrary strings as a url prefix, to which you then append a suffix. Looks like `schema:person` or `as:actor`. And these are supposed to resolve to real IRIs. And they can appear anywhere in a JSON-LD document that IRIs are valid. And they can reference each other. So `schema: https://schema.org/` and later `person: schema:person#` and finally `person:name` should actually resolve to `https://schema.org/person#name`

Basically, this spec is cursed, and I'm pretty sure it's impossible to entirely parse these documents without making at least 2 passes over the fully loaded text. The CompactIri class *does not* fully support this. It's a sort of best effort, 1 level deal. It could probably be a little better, but ughhhhhhhhhhhhhhhhhh

**Review by jenniferplusplus — APPROVED** [review]

File: Letterbook.ActivityPub/CompactIri.cs
Comment (runewake2) [reply Letterbook.ActivityPub/CompactIri.cs:1]: You're building off of Uri's in .NET which should already have [built-in IRI support](https://learn.microsoft.com/en-us/dotnet/api/system.uri?view=net-7.0#international-resource-identifier-support) in .NET Core so I *think* it should just work. Not familiar here though so IDK the limitations. I imagine we'll have a lot of situations where we build things partially and then finish and harden that implementation over time.
